src/Features/CSharp/Portable/ValidateFormatString/ValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+            else if (invocationExpr.Expression is IdentifierNameSyntax identifierNameSyntax)
+            {
+                if (identifierNameSyntax.ToString() != "Format")
+                {
+                    return;
+                }
+
+                name = identifierNameSyntax;
+            }
+            else
+            {
+                return;
+            }
+
+            var arguments = invocationExpr.ArgumentList.Arguments;
+            var numberOfArguments = arguments.Count;
+
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(name, context.CancellationToken);
+           
+            if (!TryGetFormatMethod(context, numberOfArguments, symbolInfo, out var method))
+            {
+                return;
+            }
+
+            var hasIFormatProvider =
+                method.Parameters[0].Type.ToString().Equals(typeof(IFormatProvider).FullName);
+
+            if (!TryGetFormatStringAndLocation(
+                arguments,
+                hasIFormatProvider,
+                out var formatString,
+                out var formatStringLocation))
+            {
+                return;
+            }
+
+            ValidateAndReportDiagnostic(context, hasIFormatProvider, arguments.Count, formatString, formatStringLocation);
+        }
+
+        private static bool TryGetFormatStringAndLocation(
+            SeparatedSyntaxList<ArgumentSyntax> arguments, 
+            bool hasIFormatProvider, 
+            out string formatString, 
+            out Location formatStringLocation)
+        {
+            formatString = null;
+            formatStringLocation = null;
+
+            var formatArgumentSyntax = GetFormatStringArgument(arguments, hasIFormatProvider);
+            if (formatArgumentSyntax == null)
+            {
+                return false;
+            }
+
+            if (!formatArgumentSyntax.Expression.IsKind(SyntaxKind.StringLiteralExpression))
+            {
+                return false;
+            }
+
+            var formatLiteralExpressionSyntax = (LiteralExpressionSyntax)formatArgumentSyntax.Expression;
+            formatString = formatLiteralExpressionSyntax.Token.ValueText;
+            formatStringLocation = formatArgumentSyntax.GetLocation();
+
+            return true;
+        }
+
+        private static ArgumentSyntax GetFormatStringArgument(SeparatedSyntaxList<ArgumentSyntax> arguments, bool hasIFormatProvider)
+        {
+            foreach (var argument in arguments)
+            {
+                if (argument.NameColon != null && !argument.NameColon.IsMissing && argument.NameColon.Name.Identifier.ValueText.Equals("format"))
+                {
+                    return argument;
+                }
+            }
+
+            // If using positional arguments, the format string will be the first or second
+            // argument depending on whether there Is an IFormatProvider argument.
+            return (hasIFormatProvider ? arguments[1] : arguments[0]);
```

Comment:
Does something validate that there are enough aguments to index into this?  I saw tests for the no-argument case, but I don't see why we wouldn't reach here.

Can you comment on why we know this is safe?